tests/utils.test.ts

```diff
@@ -160,6 +161,40 @@ describe('utils', () => {
       },
     );
 
+    // Test dataAttributes
+    const dataAttributes: DataAttributes[] = [
+      { name: 'test', value: 'test' },
+      { name: 'test2', value: 'test2' },
+    ];
+    test(
+      JSON.stringify({
+        compatibility: false,
+        defer: false,
+        dataAttributes,
+      }),
+      () => {
+        expect(window.dataLayer).toBeUndefined();
+        expect(document.scripts.length).toBe(0);
+
+        const script: HTMLScriptElement = loadScript('GTM-DEMO', {
+          compatibility: false,
+          defer: false,
+          dataAttributes,
+        });
+
+        expectDataLayerToBeCorrect();
+        expectScriptToBeCorrect({
+          src: 'https://www.googletagmanager.com/gtm.js?id=GTM-DEMO',
+          async: true,
+          defer: false,
+          nonce: '',
+          dataAttributes,
```

Comment:
**issue (blocking):** you are passing, but it is not used right now at all
please change the `expectScriptToBeCorrect` to test internally for data attribute correctness